var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -58,12 +58,21 @@ class Gromacs(CMakePackage):
                     'AVX2_128', 'AVX2_256', 'AVX_512', 'AVX_512_KNL',
                     'IBM_QPX', 'Sparc64_HPC_ACE', 'IBM_VMX', 'IBM_VSX',
                     'ARM_NEON', 'ARM_NEON_ASIMD'))
-    variant('rdtscp', default=True, description='Enable RDTSCP instruction usage')
+    variant('rdtscp', default=True,
+            description='Enable RDTSCP instruction usage')
+    variant('subcounters', default=False,
+            description='Enable wallcycle subcounters')
+    variant('own_fftw', default=True, description='Build own FFTW')
```

Comment:
Building an internal fftw is contrary what spack tries to do, so drop.

Reply:
As this is a performance issue, we should expose the missing option in the `fftw` spackage.

Reply:
I agree... default should be to use a Spack-provided FFTW.  If the Spack FFTW recipe doesn't use the right options, it should be fixed.

Reply:
Also, the sense of this variant should be reversed, i.e:
```
variant('external_fftw', default=True, description='Use Spack-provided FFTW (don't build it ourselves)')
```
This switch makes the dependencies below less weird.  Note that I left `default=True`.  Default with Spack should always be to use other Spack-built packages.

Reply:
@junghans and @citibeth, does it make sense to, on a Cray, to have this option?
Because there, people map the `fftw` to the cray provided module. In this case, building it own version adds performance. Or do you have any other suggestions?

Reply:
I think it's fine to have an option to build an internal FFTW if that's something GROMACS supports and it makes sense to experiment with it.  If you want to build with the Cray FFTW, you can add it to your external packages and build with `+external_fftw`.  I also agree we should set the default to just depend on `fftw` in Spack.

Reply:
The other problem with this options that is used CMake's `ExternalProject_add()`, which downloads stuff at configure time, so this option won't work with spack mirrors.

Looking at https://github.com/gromacs/gromacs/blob/master/src/external/build-fftw/CMakeLists.txt#L88 there is really nothing magical about the internal fftw, so it seems `depends_on('fftw simd=sse2,avx,avx2')` should do the same.

Reply:
Ok, if this *downloads* a separate FFTW, then we shouldn’t add it at all, unless we can fetch the archive ourselves with a `resource` directive.  Packages should not be downloading their own stuff without spack knowing.

Reply:
> @junghans and @citibeth, does it make sense, on a Cray, to have this option?
> Because there, people map the `fftw` to the cray provided module. In this case, building it own version adds performance. Or do you have any other suggestions?

I don't know, I'm not familiar with  Cray.  But Spack needs to work for other platforms too.